functions/signals/ks_main.php

```diff
@@ -108,45 +112,43 @@ public static function generateImage($height)
 			}
 		}
 		
+		$radius_main_lamp = 3.5;
+		$radius_small_lamp = 2.25;
+		
 		$image = '
 			<g transform="translate(0 ' . $height . ')">
 				<g>
-					<polygon style="&background;" points="6,1 34,1 34,59 6,59"/>
+					<polygon style="&background;" points="6,1 ' . $width-6 . ',1 ' . $width-6 . ',' . $height+1 . ' 6,' . $height+1 . '"/>
 				</g>
 					
 				<g id="hp0">
-					<circle style="' . $colour_hp0 . '" cx="20" cy="16" r="4"/>
+					<circle style="' . $colour_hp0 . '" cx="' . $width/2 . '" cy="16" r="' . $radius_main_lamp . '"/>
 				</g>
 					
 				<g id="ks1">
-					<circle style="' . $colour_ks1 . '" cx="20" cy="29" r="4"/>
+					<circle style="' . $colour_ks1 . '" cx="' . $width/2 . '" cy="29" r="' . $radius_main_lamp . '"/>
 				</g>
 				<g id="sh1">';
 		if ( ( isset($_GET["railway:signal:main:substitute_signal"]) && $_GET["railway:signal:main:substitute_signal"] == "DE-ESO:dr:zs1" ) || ( isset($_GET["railway:signal:minor"]) && $_GET["railway:signal:minor"] == "DE-ESO:sh1" ) )
 		{
 			$image .= '
-					<circle style="&gray;" cx="20" cy="39" r="2"/>
-					<circle class="' . $class_zs1 . '" style="' . $colour_zs1 . '" cx="20" cy="39" r="2"/>';
+					<circle style="&gray;" cx="' . $width/2 . '" cy="44" r="' . $radius_small_lamp . '"/>
+					<circle class="' . $class_zs1 . '" style="' . $colour_zs1 . '" cx="' . $width/2 . '" cy="39" r="' . $radius_small_lamp . '"/>';
```

Comment:
Both lights needs to be at the same spot, so that the grey background is still shown when the light blinks.